packages/rollup-tests/test/function/index.js

```diff
@@ -24,24 +24,36 @@ function requireWithContext(code, context, exports) {
 		error.exports = module.exports;
 		throw error;
 	}
-	return contextWithExports.module.exports;
+	const moduleExports = contextWithExports.module.exports;
+	if (isImport) {
+		return { default: moduleExports, ...moduleExports };
+	}
+	return moduleExports;
 }
 
-function runCodeSplitTest(codeMap, entryId, configContext) {
+function runCodeSplitTest(codeMap, entryId, configContext, dynamicImportInCjs) {
 	const exportsMap = Object.create(null);
 
-	const requireFromOutputVia = importer => importee => {
+	const requireFromOutputVia = (importer, isImport) => importee => {
 		const outputId = path.posix.join(path.posix.dirname(importer), importee);
 		if (outputId in exportsMap) {
 			return exportsMap[outputId];
 		}
 		const code = codeMap[outputId];
 		return code === undefined
-			? require(importee)
+			? isImport ? import(importee) : require(importee)
 			: (exportsMap[outputId] = requireWithContext(
-					code,
-					{ require: requireFromOutputVia(outputId), ...context },
-					(exportsMap[outputId] = {})
+				// Here replaced `import` to avoid unexpected error
```

Comment:
I recommended you split the rewrite logic in another file,this could make it more clear for the tests runner logic, especially when you got lot fo rewrite rules.

Reply:
Here split the rewrite logic need to copy functions, it look like is unnecessary at now.